cmd/payledger/payledger.go

```diff
@@ -229,21 +230,23 @@ func buildLedger(node INode, config *Config) error {
 				}
 			}
 			if _, ok := genesisLedger[addrStr]; !ok {
-				tp := ledger.TokenPayout{Address: addrStr, PkScript: entry.PkScript(), Amount: 0}
-				reTp := ledger.TokenPayoutReGen{tp, 0}
+				tp := ledger.TokenPayout{Address: addrStr, PkScript: entry.PkScript(), Amount: types.Amount{Value: 0, Id: types.MEERID}}
+				reTp := ledger.TokenPayoutReGen{tp, types.Amount{Value: 0, Id: types.MEERID}}
 				genesisLedger[addrStr] = &reTp
 			}
 
 			if params.GenesisHash.IsEqual(entry.BlockHash()) {
-				genesisLedger[addrStr].GenAmount += entry.Amount()
-				genAmount += entry.Amount()
+				if genesisLedger[addrStr].GenAmount.Id == entry.Amount().Id {
+					genesisLedger[addrStr].GenAmount.Value += entry.Amount().Value
+					genAmount += uint64(entry.Amount().Value)
+				}
 			} else {
 				eamount := entry.Amount()
```

Comment:
eamount looks a little bit ugly to me, I like eAmount

Reply:
ok , BTW, Warning... please don't rely on `payledger` cmd, it should  be rewritten totally in the future. at current stage. I only had fixed all compile errors, but don't check the internal logic very carefully.

Reply:
OK